addons/hw_drivers/iot_handlers/drivers/L10nKeEDISerialDriver.py

```diff
@@ -149,8 +159,14 @@ def send(self, msgs):
                 request = struct.pack('B%ds2sB' % (len(core_message)), STX, core_message, self.generate_checksum(core_message), ETX)
                 time.sleep(self._protocol.commandDelay)
                 self._connection.write(request)
-                time.sleep(self._protocol.measureDelay)
-                response = self._connection.read_all()
+                # If we know the expected output size, we can set the read
+                # buffer to match the size of the output.
+                output_size = COMMAND_OUTPUT_SIZE.get(msg[0])
+                if output_size:
+                    response = self._connection.read(COMMAND_OUTPUT_SIZE.get(msg[0]))
+                else:
+                    time.sleep(self._protocol.measureDelay)
+                    response = self._connection.read_all()
```

Comment:
Why not use `read_all` for all response ?
what is the advantage of reading exactly only the size of the response in this case ?

Reply:
Because you can not skip the measuredelay then.